### PR TITLE
HARVESTER: Better loadblancer config of Harvester cloud provider

### PR DESCRIPTION
**Pull request by n313893254**
- https://github.com/rancher/dashboard/issues/4451

### CODE REVIEW

File: utils/cluster.js [hunk 1]
```diff
@@ -15,3 +16,13 @@ export function isHarvesterCluster(mgmtCluster) {
 
   return provider === VIRTUAL_HARVESTER_PROVIDER;
 }
+
+export function isHarvesterSatisfiesVersion(version = '') {
```
Comment (richard-cox) [inline utils/cluster.js:20]: This should be called something more generic, as it's applied in all cases

Comment (richard-cox) [reply utils/cluster.js:20]: This looks like it's changed the natural flow for non-harvester clusters. Could you explain the change?

**Review by richard-cox — CHANGES_REQUESTED** [review]
I haven't looked at this thoroughly yet, but had some questions regarding `isHarvesterSatisfiesVersion`

File: utils/cluster.js (hunk 1 above)
Comment (n313893254) [reply utils/cluster.js:20]: Create a Harvester cluster requires RKE version >= `v1.21.4+rke2r4`. So I need to init k8s version when select create a harvester cluster

```js
      if (this.isHarvesterDriver && showRke2) {
        this.setHarvesterK8sDefaultVersion();
      } else {
        set(this.value.spec, 'kubernetesVersion', option.value);
      }
```

**Comment by guangbochen** [comment]
@richard-cox this PR should be ready for review, and please let us know if there is anything need to be changed, thanks.

File: utils/cluster.js (hunk 1 above)
Comment (richard-cox) [reply utils/cluster.js:20]: Sorry, i hadn't spotted this fn was only ever called in Harvester world

File: edit/service.vue [hunk 2]
```diff
@@ -75,6 +107,12 @@ export default {
       total:   0,
     };
 
+    const harvesterAddOnConfig = {};
```
Comment (richard-cox) [inline edit/service.vue:110]: Can the harvester specific code here be gated on `showHarvesterAddOnConfig` (might need to move this into the fetch)?

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: edit/service.vue (hunk 2 above)
Comment (n313893254) [reply edit/service.vue]: Done. I add a component named "HarvesterServiceAddOnConfig". When showHarvesterAddOnConfig is true, shows the Add-on config.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**